core-it-suite/src/test/java/org/apache/maven/it/IntegrationTestSuite.java

```diff
@@ -106,6 +106,10 @@ public static Test suite()
         // Tests that don't run stable and need to be fixed
         // -------------------------------------------------------------------------------------------------------------
         // suite.addTestSuite( MavenIT0108SnapshotUpdateTest.class ); -- MNG-3137
+        suite.addTestSuite( MavenITmng4106InterpolationUsesDominantProfile400Test.class );
```

Comment:
I am a bit confused by this. Some of the are already there, why they pop up here now and why have some disabled?

Reply:
The original tests fail starting with the introduced optional behavior in Maven 4, that's why we copied the original testcase and slightly adjusted them (and appended 400 to the class name, meaning version 4.0.0).

We have done this as well for the MavenITmng4262MakeLikeReactorDottedPathTest testcase in another feature on @rfscholte's advice.

Reply:
Hmm, may ITs have an inline if for those cases. Why not use that style? It reduces the amount of code we have. Example: https://github.com/apache/maven-integration-testing/pull/96/files#diff-0323fd1cd7d6126ff8efaf53b1771bb88deb78c458141d38695979480c6710e5R58

Reply:
My opinion is that those ifs can be useful in the tests when only a single thing has to be changed per version. 
In the classes that we have duplicated we noticed that we would have to change the `.addCliOption` (which activates the profiles) everywhere and that these tests already contained big `if (mvnVersion < 3) `-branches which tested support for profiles.xml.

So in this case my preference is to duplicate the test class to keep the tests readable and immutable (which was @rfscholte 's advice).

It does increase the amount of code + tests indeed, but I don't think that outweighs the benefits listed above. 
What do you think?

Reply:
My general rule of thumb is if the test has a different structure or requires 50+% in code modification then it needs to be copied. I need to take a closer look how the diff actually looks like.